src/index.ts

```diff
@@ -16,6 +16,12 @@ export default function electron(config: Configuration | Configuration[]): Plugi
         server.httpServer?.once('listening', () => {
           bootstrap(config, server)
         })
+        server.httpServer?.once('close', () => {
+          if (process.electronApp) {
+            process.electronApp.removeAllListeners()
+            process.electronApp.kill()
+          }
+        })
```

Comment:
Maybe it's not necessary!
https://github.com/vitejs/vite/blob/060cd66878eb6bee53a55ea3ec0afd724ee9b35c/packages/vite/src/node/shortcuts.ts#L119
<img width="458" alt="image" src="https://user-images.githubusercontent.com/26263658/207210195-44dd17df-31ba-49d6-8c85-db09389c1ad1.png">
Eventually, the process will exit after the server is close.

Reply:
With Vite 4, using ctrl+c without listening for the close event, the Electron process won’t get killed.

So maybe it’s not required for the quit action… But it is for ctrl+c

Reply:
I used Vite4 in [electron-vite-vue](https://github.com/electron-vite/electron-vite-vue), I tried `Ctrl+C` and Electron exited normally.

Reply:
Hahaha, yes, you are right 😆

Reply:
Happy to contribute, thanks for this project.